### ISSUE TITLE
Fees: investigate and implement fee-burning

### ISSUE THREAD

**Issue by liamsi** [celestiaorg/celestia-app#49]
We want to implement eip-1559 style fee burning (but in the current tendermint deferred execution model).

This requires two parts:

1. Investigate how to achieve the above using the current execution model and within the existing ABCI methods (keeping in mind that abci++ is currently being spec'ed out). This needs to be summarized in an ADR and @adlerjohn volunteered to have a look into this.
2. Implement the fee-burning mechanism according to the above ADR in alignment with the specs.

Until then, we can probably just live with #48.

**Comment by hxrts** [comment]
@fedekunze has done some work on this if you didn't catch his comment in discord https://github.com/celestiaorg/lazyledger-app/issues/49

**Comment by liamsi** [comment]
Yeah, @fedekunze mentioned that to me but I didn't see the discord message nor the code so far. I'll check it out! Thanks for the pointer @hxrts 🙏🏼

**Comment by Wondertan** [comment]
Yet an unpopular [concern](https://mtefagh.github.io/fee/) about EIP-1559 with in-depth analysis. It might be worth looking into.

**Comment by evan-forbes** [comment]
closed with #380

**Comment by evan-forbes** [comment]
just for posterity: as we've discussed in other issues/channles, we cannot have fee burning until we also have immediate execution.

**Comment by evan-forbes** [comment]
> just for posterity: as we've discussed in other issues/channles, we cannot have fee burning until we also have immediate execution.

We don't actually need immediate execution, but we do need to ensure that each transaction that has fee burning (only PFBs in our case, can be executed properly when it gets included in a block. We are already doing a lot of this do our existing block validity rules, the only rules that we would have to add are:
- [ ] #979
- [ ] Signers of PFBs have enough funds to pay for the fee of their transaction
- [ ] Enforce a minimum fee

**Comment by evan-forbes** [comment]
ref https://github.com/celestiaorg/celestia-app/issues/1971
